### PR TITLE
Various fixes for v4 docs

### PR DESCRIPTION
**Pull request by DomFijan**
## Description

Several examples at various places in the docs were incorrect (md example in readme, new thermostat examples etc.). I've also updated some thermostat docstrings with additional examples. 

## Motivation and context

Hopefully makes docs better.

## How has this been tested?

Docs build successfully in tests and all changes were visually inspected.

## Change log

```
Updated docstrings for new integration methods.
Updated examples on documentation front page and readme.
```

## Checklist:

- [x] I have reviewed the [**Contributor Guidelines**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/CONTRIBUTING.rst).
- [x] I agree with the terms of the [**HOOMD-blue Contributor Agreement**](https://github.com/glotzerlab/hoomd-blue/blob/trunk-minor/ContributorAgreement.md).
- [x] My name is on the list of contributors (`sphinx-doc/credits.rst`) in the pull request source branch.

### CODE REVIEW

File: hoomd/md/methods/methods.py [hunk 1]
```diff
@@ -283,22 +286,36 @@ class ConstantPressure(Thermostatted):
...
+
+        # NPT integrator with cubic symmetry
         npt = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
         tauS = 1.2, S=2.0, couple="xyz",
         thermostat=hoomd.md.methods.thermostats.Bussi(kT=1.0))
-        # orthorhombic symmetry
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[npt], forces=[lj])
+
+        # NPT integrator with orthorhombic symmetry
         nph = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```
Comment (b-butler) [inline hoomd/md/methods/methods.py]: should be npt
```suggestion
        npt = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```

File: hoomd/md/methods/methods.py [hunk 2]
```diff
@@ -283,22 +286,36 @@ class ConstantPressure(Thermostatted):
...
-        tauS = 1.2, S=2.0, couple="none")
-        # tetragonal symmetry
+        tauS = 1.2, S=2.0, couple="none",
+        thermostat=hoomd.md.methods.thermostats.Bussi(kT=1.0))
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[npt], forces=[lj])
+
+        # NPT integrator with tetragonal symmetry
         nph = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```
Comment (b-butler) [inline hoomd/md/methods/methods.py]:
```suggestion
        npt = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```

File: hoomd/md/methods/methods.py [hunk 3]
```diff
@@ -283,22 +286,36 @@ class ConstantPressure(Thermostatted):
...
-        tauS = 1.2, S=2.0, couple="xy")
-        # triclinic symmetry
+        tauS = 1.2, S=2.0, couple="xy",
+        thermostat=hoomd.md.methods.thermostats.Bussi(kT=1.0))
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[npt], forces=[lj])
+
+        # NPT integrator with triclinic symmetry
         nph = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```
Comment (b-butler) [inline hoomd/md/methods/methods.py]:
```suggestion
        npt = hoomd.md.methods.ConstantPressure(filter=hoomd.filter.All(),
```

File: hoomd/md/methods/thermostats.py [hunk 4]
```diff
@@ -62,11 +62,26 @@ class MTTK(Thermostat):
...
         <http://dx.doi.org/10.1063/1.470959>`_.
 
+    Example::
+
+        # NVT integration using MTTK Nose-Hoover thermostat
+        dt = 0.005
+        MTTK = hoomd.md.methods.thermostats.Berendsen(kT=1.0, tau=dt*100)
+        nvt = hoomd.md.methods.ConstantVolume(filter=hoomd.filter.All(),
+                                            thermostat=MTTK)
+        integrator = hoomd.md.Integrator(dt=dt, methods=[nvt], forces=[lj])
```
Comment (b-butler) [inline hoomd/md/methods/thermostats.py]: There are no forces defined. Perhaps we should remove that part of the example.
```suggestion
        integrator = hoomd.md.Integrator(dt=dt, methods=[nvt])
```

File: hoomd/md/methods/thermostats.py [hunk 5]
```diff
@@ -147,6 +162,13 @@ class Bussi(Thermostat):
     See Also:
         `Bussi et. al. 2007 <https://doi.org/10.1063/1.2408420>`_.
 
+    Example::
+
+        # NVT integration using Bussi thermostat
+        bussi = hoomd.md.methods.thermostats.Bussi(kT=1.0)
+        nvt = hoomd.md.methods.ConstantVolume(filter=hoomd.filter.All(),
+                                            thermostat=bussi)
+        integrator = hoomd.md.Integrator(dt=0.005, methods=[nvt], forces=[lj])
```
Comment (b-butler) [inline hoomd/md/methods/thermostats.py]:
```suggestion
        integrator = hoomd.md.Integrator(dt=0.005, methods=[nvt])
```

File: hoomd/md/methods/thermostats.py [hunk 6]
```diff
@@ -186,6 +208,15 @@ class Berendsen(Thermostat):
...
         `Berendsen et. al. 1984 <http://dx.doi.org/10.1063/1.448118>`_.
 
+    Example::
+
+        # NVT integration using Berendsen thermostat
+        dt = 0.005
+        berendsen = hoomd.md.methods.thermostats.Berendsen(kT=1.0, tau=dt*100)
+        nvt = hoomd.md.methods.ConstantVolume(filter=hoomd.filter.All(),
+                                            thermostat=berendsen)
+        integrator = hoomd.md.Integrator(dt=dt, methods=[nvt], forces=[lj])
```
Comment (b-butler) [inline hoomd/md/methods/thermostats.py]:
```suggestion
        integrator = hoomd.md.Integrator(dt=dt, methods=[nvt])
```

**Review by b-butler — CHANGES_REQUESTED** [review]
Thanks @DomFijan for the fixes. I have made some suggestions.

**Comment by DomFijan** [comment]
@b-butler thanks for this. At last dev meeting it was mentioned (not sure by whom, perhaps @joaander ) that preferably examples should be such that they can be copy-pasted, since users tend to copy paste these examples.
In this light, do you think it would be better to complement the example with, like:
```
nl = nlist.Cell()
lj = pair.LJ(nl, default_r_cut=3.0)
```
or would you be in favor of getting rid of `forces = [lj]` as you outlined in the suggestion?

**Comment by joaander** [comment]
> @b-butler thanks for this. At last dev meeting it was mentioned (not sure by whom, perhaps @joaander ) that preferably examples should be such that they can be copy-pasted, since users tend to copy paste these examples. In this light, do you think it would be better to complement the example with, like:
> 
> ```
> nl = nlist.Cell()
> lj = pair.LJ(nl, default_r_cut=3.0)
> ```
> 
> or would you be in favor of getting rid of `forces = [lj]` as you outlined in the suggestion?

Yes, I mentioned this. There is an argument for both. Without `forces = [lj]`, the user can copy and paste any of these blocks and they will work provided they have `integrator.forces = [force1, force2, ...]` elsewhere in their code.

On the other hand, with the neighbor list and lj boilerplate, the example is 2 lines closer to being a complete copyable example. Users would need to provide `lj.params` for all their particle types. I defer to users to decide which is more better.

**Comment by b-butler** [comment]
I prefer without LJ as I would rather not assume things of the users' needs. Also, like @joaander said it would still require further specification, and technically running without forces is not an error.

**Comment by DomFijan** [comment]
Makes sense. Removed.

**Review by b-butler — APPROVED** [review]

**Pull request merged**